src/web/server/document.tsx

```diff
@@ -173,22 +178,33 @@ export const document = ({ data }: Props): string => {
 	const polyfillIO =
 		'https://assets.guim.co.uk/polyfill.io/v3/polyfill.min.js?rum=0&features=es6,es7,es2017,es2018,es2019,default-3.6,HTMLPictureElement,IntersectionObserver,IntersectionObserverEntry,fetch,NodeList.prototype.forEach&flags=gated&callback=guardianPolyfilled&unknown=polyfill&cacheClear=1';
 
+	const pageHasInteractiveElements = CAPIElements.some(
+		(element) =>
+			element._type ===
+			'model.dotcomrendering.pageElements.InteractiveBlockElement',
+	);
 	/**
 	 * The highest priority scripts.
 	 * These scripts have a considerable impact on site performance.
 	 * Only scripts critical to application execution may go in here.
 	 * Please talk to the dotcom platform team before adding more.
 	 * Scripts will be executed in the order they appear in this array
 	 */
+	function isDefined<T>(argument: T | boolean): argument is T {
+		return argument !== false;
+	}
 	const priorityScriptTags = generateScriptTags(
 		[
 			{ src: polyfillIO },
 			...getScriptArrayFromChunkName('ophan'),
 			CAPI.config && { src: CAPI.config.commercialBundleUrl },
 			...getScriptArrayFromChunkName('sentryLoader'),
 			...getScriptArrayFromChunkName('dynamicImport'),
+			pageHasInteractiveElements && {
+				src: `${CDN}static/frontend/js/curl-with-js-and-domReady.js`,
+			},
 			...arrayOfLoadableScriptObjects, // This includes the 'react' entry point
-		].filter(Boolean),
+		].filter(isDefined), // We use the TypeGuard to keep TS happy
```

Comment:
If we inline this does that make the code simpler or add a bunch of ts issues?

Reply:
I think this is a rare occasion where inlining reduced comprehension because we lose the function name that specifically helps with the comprehension